src/crypto/CMakeLists.txt

```diff
@@ -0,0 +1,105 @@
+# Copyright (c) 2023 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+if(ASM AND NOT MSVC)
+  # Check for SSE4.1 intrinsics.
+  set(SSE41_CXXFLAGS -msse4.1)
+  check_cxx_source_compiles_with_flags("${SSE41_CXXFLAGS}" "
+    #include <immintrin.h>
+
+    int main()
+    {
+      __m128i l = _mm_set1_epi32(0);
+      return _mm_extract_epi32(l, 3);
+    }
+    " HAVE_SSE41
+  )
+
+  # Check for AVX2 intrinsics.
+  set(AVX2_CXXFLAGS -mavx -mavx2)
+  check_cxx_source_compiles_with_flags("${AVX2_CXXFLAGS}" "
+    #include <immintrin.h>
+
+    int main()
+    {
+      __m256i l = _mm256_set1_epi32(0);
+      return _mm256_extract_epi32(l, 7);
+    }
+    " HAVE_AVX2
+  )
+
+  # Check for x86 SHA-NI intrinsics.
+  set(X86_SHANI_CXXFLAGS -msse4 -msha)
+  check_cxx_source_compiles_with_flags("${X86_SHANI_CXXFLAGS}" "
+    #include <immintrin.h>
+
+    int main()
+    {
+      __m128i i = _mm_set1_epi32(0);
+      __m128i j = _mm_set1_epi32(1);
```

Comment:
Not that it matters for the outcome of the check, but defining and not using `j` looks weird.

Reply:
Pull from `configure.ac`.

I guess, it was `return _mm_extract_epi32(_mm_sha256rnds2_epu32(i, j, k), 0);` assumed :)

Reply:
> Not that it matters for the outcome of the check, but defining and not using `j` looks weird.

Thanks! [Updated](https://github.com/hebasto/bitcoin/pull/7#issuecomment-1443742611).